components/director/pkg/persistence/sql_error_mapper.go

```diff
@@ -17,18 +18,23 @@ func MapSQLError(err error, resourceType resource.Type, format string, args ...i
 	}
 
 	if err == sql.ErrNoRows {
+		log.Warnf("SQL: no rows in result set for '%s' table", resourceType)
 		return apperrors.NewNotFoundErrorWithType(resourceType)
 	}
 
 	pgErr, ok := err.(*pq.Error)
 	if !ok {
+		log.Warn("Error while getting postgres error code")
 		return apperrors.InternalErrorFrom(err, format, args...)
 	}
 
+	log.Debug("Checking Postgres error code...")
 	switch pgErr.Code {
 	case UniqueViolation:
+		log.Warn("Postgres unique violation error code found")
```

Comment:
You can add more context in the logs, resource type and arguments and the error itself this does not give much more value than the error returned on the next line.